symphony/app/fbcnms-projects/hub/app/components/workflow/pages/workflowList/Scheduling/Scheduling.js

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from "react";
+import {
+  Accordion,
+  Button,
+  Card,
+  Col,
+  Container,
+  Row,
+  Table
+} from "react-bootstrap";
+import { withRouter } from "react-router-dom";
+import PageCount from "../../../common/PageCount";
+import PageSelect from "../../../common/PageSelect";
+import { conductorApiUrlPrefix, frontendUrlPrefix } from "../../../constants";
+import SchedulingModal from "./SchedulingModal/SchedulingModal";
+import superagent from "superagent";
+
+const Scheduling = props => {
+  const [showSchedulingModal, setShowSchedulingModal] = useState(false);
+  const [activeRow, setActiveRow] = useState();
+
+  const refresh = () => {
+    const path = conductorApiUrlPrefix + "/schedule/";
+    const req = superagent.get(path).accept("application/json");
+    req.end((err, res) => {
+      if (res && res.ok && Array.isArray(res.body)) {
+        const result = res.body;
+        let size = Math.floor(result.length / defaultPages);
+        let dataset = result.sort((a, b) =>
+            a.name > b.name ? 1 : b.name > a.name ? -1 : 0
+          ) || [];
+        setData(dataset);
+        setPagesCount(result.length % defaultPages ? ++size : size);
+        deselectActiveRow();
+      } else {
+        // TODO
+      }
+    });
+    return [];
+  };
+
+  const [data, setData] = useState(refresh);
```

Comment:
Could you clarify what's happening here? refresh is a lambda, how is it used as "state" here? Also, I'm seeing `data` and `setData` initialized here using refresh, but the refresh lambda references `setData` (which seems to me like a circular dependency). Just need some clarity here. Thanks.

Reply:
I've struggled there trying to use [Lazy initial state](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state) however I needed the function result to be a promise and that does not work, so I have returned [] and did the actual state initialization in the lambda.
After your comment I've read an [article](https://codeburst.io/master-the-asynchronous-state-in-a-functional-react-component-60a0899cffca) about async & functional components and modified the initialization according to it. Not sure which approach is preferable, I will be happy to redo it if there is a more idiomatic way.

Reply:
Neat, I read the article and that looks like a good solution to me 👍 thanks, will approve the PR